File: src/fizzbuzz/fizz_buzz.rb
# frozen_string_literal: true

# FizzBuzz :
class FizzBuzz
  def self.iterate(num)
    res = ''
    (1..num).each do |n|
      res += fizz_buzz(n)
      res += ', ' if n != num
    end
    res += "\n"
  end

  def self.fizz_buzz(num)
    if (num % 15).zero?
      'Fizz Buzz'
    elsif (num % 3).zero?
      'Fizz'
    elsif (num % 5).zero?
      'Buzz'
    else
      num.to_s
    end
  end
end

File: rust/src/fizzbuzz/fizz_buzz.rs
pub fn iterate(num: i32) -> String {
  let mut res = "".to_string();

  for n in 1..=num {
    if n % 15 == 0 {
      res.push_str("Fizz Buzz");
    } else if n % 3 == 0 {
      res.push_str("Fizz");
    } else if n % 5 == 0 {
      res.push_str("Buzz");
    } else {
      res = format!("{}{}", res, n)
    }
    if n != num {
      res.push_str(", ");
    }
  }
  res.push_str("\n");
  res
}
